File: tests/playwright/tests/TS01.js
const { test, expect } = require('@playwright/test');

test('TS01 – Platná registrace uživatele', async ({ page }) => {
  // 1. Otevření registrační stránky
  await page.goto('https://example.com/register');

  // 2. Vyplnění formuláře
  await page.fill('input[name="full_name"]', 'Don Salieri');
  await page.fill('input[name="user_name"]', 'DonS');
  await page.fill('input[name="password"]', 'P.lb.45_?1!');

  // 3. Kliknutí na tlačítko "Register"
  await page.click('input[type="submit"]');

  // 4. Ověření, že se zobrazí zpráva o úspěšné registraci
  const successMessage = await page.locator('text=Registration successful');
  await expect(successMessage).toBeVisible();

  console.log('Registrace byla úspěšná .... [PASS]');
});

File: tests/selenium/TS01.py
from BaseTest import BaseTest
import logging

class TS01(BaseTest):

    def test_register_valid_user(self):
        logging.info("=============================================")
        logging.info("Test TS01:User registration to the system begins.")
        logging.info("=============================================")
        self.openurl("http://localhost/spravce/public/register.php")
        self.fill_text("full_name", "DonSalieri")
        self.fill_text("user_name", "DonS")
        self.fill_text("password", "P.lb.45_?1!")
        self.click_button("input[type='submit']", "success")
        
if __name__ == "__main__":
    import unittest
    unittest.main()
